llvm/lib/CASObjectFormats/SchemaBase.cpp

```diff
@@ -7,8 +7,30 @@
 //===----------------------------------------------------------------------===//
 
 #include "llvm/CASObjectFormats/SchemaBase.h"
+#include "llvm/CASObjectFormats/FlatV1.h"
+#include "llvm/CASObjectFormats/NestedV1.h"
 
 using namespace llvm;
 using namespace llvm::casobjectformats;
 
 void SchemaBase::anchor() {}
+
+Expected<std::unique_ptr<SchemaBase>> SchemaBase::create(cas::CASDB &CAS,
+                                                         cas::CASID ID) {
+  Expected<cas::NodeRef> Ref = CAS.getNode(ID);
+  if (!Ref)
+    return Ref.takeError();
+  {
+    auto Schema = std::make_unique<flatv1::ObjectFileSchema>(CAS);
+    if (Schema->isRootNode(*Ref))
+      return Schema;
+  }
+  {
+    auto Schema = std::make_unique<nestedv1::ObjectFileSchema>(CAS);
+    if (Schema->isRootNode(*Ref))
+      return Schema;
+  }
```

Comment:
This is duplicating code from `llvm-cas-object-format`. Can you add a FIXME over there to use the new API?